src/is-package-changed.ts

```diff
@@ -30,18 +33,21 @@ async function isPackageChanged(
     options: PackageChangedOptions = {},
     callback?: PackageChangedCallback,
 ): Promise<PackageChangedResult | Omit<PackageChangedResult, 'writeHash'>> {
-    const { hashFilename = '.packagehash', cwd = process.cwd() } = options;
+    const { hashFilename = '.packagehash', cwd = process.cwd(), lockfile } = options;
     const packagePath = findPackage({ cwd });
     if (!packagePath) {
         throw new Error('Cannot find package.json. Travelling up from current working directory.');
     }
+    const packagelockPath = findPackagelock({ cwd });
 
     const packageHashPath = path.join(cwd, hashFilename);
     const writeHash = (hash: string | undefined) =>
         hash && fs.writeFileSync(packageHashPath, hash, {});
 
     const packageHashPathExists = fs.existsSync(packageHashPath);
-    const recentDigest = getPackageHash(packagePath);
+    const recentDigest = lockfile
+        ? `${getPackageHash(packagePath)}${getPackagelockHash(packagelockPath) ?? ''}`
```

Comment:
I would assume the `package-lock,json` file  contains all information needed and it would be no longer needed to get the hash from dependencies listed in `package.json` when `lockfile=true`.

Reply:
Can you explain this?